packages/trader/test/UniswapV3Trader.ts

```diff
@@ -0,0 +1,427 @@
+import { web3, assert } from "hardhat";
+
+const { toWei, fromWei } = web3.utils;
+
+const { getTruffleContract } = require("@uma/core");
+
+const {
+  encodePriceSqrt,
+  getTickFromPrice,
+  getCurrentPrice,
+  encodePath,
+  computePoolAddress,
+  FeeAmount,
+  createContractObjectFromJson,
+  replaceLibraryBindingReferenceInArtitifact
+} = require("@uma/common");
+
+// Script to test
+const { RangeTrader } = require("../src/RangeTrader");
+
+// Helper scripts
+const { createExchangeAdapter } = require("../src/exchange-adapters/CreateExchangeAdapter");
+
+const Token = getTruffleContract("ExpandedERC20", web3);
+const WETH9 = getTruffleContract("WETH9", web3);
+const DSProxyFactory = getTruffleContract("DSProxyFactory", web3, "latest");
+const DSProxy = getTruffleContract("DSProxy", web3, "latest");
+
+// Import all the uniswap related contracts.
+const SwapRouter = require("@uniswap/v3-periphery/artifacts/contracts/SwapRouter.sol/SwapRouter.json");
+const UniswapV3Factory = require("@uniswap/v3-core/artifacts/contracts/UniswapV3Factory.sol/UniswapV3Factory.json");
+const NonfungiblePositionManager = require("@uniswap/v3-periphery/artifacts/contracts/NonfungiblePositionManager.sol/NonfungiblePositionManager.json");
+const UniswapV3Pool = require("@uniswap/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json");
```

Comment:
I _think_ we can also use typescript imports for all json files (and get nice type autocomplete etc!) if we add the resolveJsonModule compiler option: https://mariusschulz.com/blog/importing-json-modules-in-typescript.

Reply:
this did not work perfectly out the box:
![image](https://user-images.githubusercontent.com/12886084/118090763-3b06d000-b3ca-11eb-8414-4ac6bd243964.png)

after adding `"resolveJsonModule": true` to the ts config it worked! great call out!